scala/scala.bzl

```diff
@@ -96,15 +96,17 @@ def _build_nosrc_jar(ctx, buildijar):
     cmd = """
 rm -f {jar_output}
 {zipper} c {jar_output} @{path}
+touch {statsfile}
```

Comment:
I'm sorry for being too iterative and I appreciate your patience but I just re-read this and wanted to ask- you're doing this (`touch`) since sometimes (only java sources / no sources at all) scala compilation doesn't happen and bazel complained otherwise? I think there's a better way to solve it by only adding it to the `outs`if we need it but since I don't want to complicate you (even more ;) ) I suggest a comment on why we're `touch`ing here?
I just think that it's not that idiomatic and so might be unclear down the line

Reply:
No problem. Scala targets that have no source files get short circuited and don't run the scalac invoker. Creating the statsfile regardless of the number of source files makes it easier to write rules/macros for generating downstream metrics related targets, IMHO.

Reply:
Ok, sounds good. 
If you think `touch` is enough for these rules/macros, I'm good.
WDYT about adding a comment to explain why we `touch`?

Reply:
I added a brief comment 🙂. Let me know if you want a more thorough one.